### PR TITLE
Deployment UX improvements

### PR DESCRIPTION
**Pull request by micholas**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

When using command palate to deploy it disappears immediately with no feedback. For large deployments the delay can be many minutes leaving the user wondering if the extension even tried to deploy the SAM app. Now it provides output indicating packaging is occurring, deployment is occurring, then success or failure.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by micholas** [comment]
image

**Comment by micholas** [comment]
image

File: src/lambda/commands/deploySamApplication.ts [hunk 1]
```diff
@@ -26,19 +33,34 @@ export async function deploySamApplication(
 
     const tempFolder = await mkdtemp('samDeploy')
     const outputTemplatePath = path.join(tempFolder, 'template.yaml')
-
+    let stage = 'packaging'
     try {
         const packageInvocation = new SamCliPackageInvocation(template.fsPath, outputTemplatePath, s3Bucket, invoker)
+        restParams.outputChannel.show(true)
+        // TODO: Add nls support
+        restParams.outputChannel.appendLine(`Packaging SAM app to "${s3Bucket}" S3 bucket`)
```
Comment (awschristou) [inline src/lambda/commands/deploySamApplication.ts]: Minor: Consider: 
`Packaging SAM Application to S3 Bucket: ${s3Bucket}`

File: src/lambda/commands/deploySamApplication.ts [hunk 2]
```diff
@@ -26,19 +33,34 @@ export async function deploySamApplication(
...
         await packageInvocation.execute()
 
+        stage = 'deploying'
         const deployInvocation = new SamCliDeployInvocation(outputTemplatePath, stackName, invoker)
+        // Deploying can take a very long time for Python Lambda's with native dependencies so user needs feedback
+        restParams.outputChannel.appendLine(localize(
+          'AWS.samcli.deploy.stackName.initiated',
+          'Deploying "{0}" stack...',
```
Comment (awschristou) [inline src/lambda/commands/deploySamApplication.ts]: Minor: Consider: 
`Deploying SAM Application to CloudFormation Stack: {0}`

File: src/lambda/commands/deploySamApplication.ts [hunk 3]
```diff
@@ -26,19 +33,34 @@ export async function deploySamApplication(
...
+          stackName
+        ))
         await deployInvocation.execute()
-
+        // TODO: Add nls support
+        const msg = `Successfully deployed "${stackName}" to S3 "${s3Bucket}"`
```
Comment (awschristou) [inline src/lambda/commands/deploySamApplication.ts]: Minor: Consider: 
`Successfully deployed SAM Application to CloudFormation Stack: ${stackName}`

At that point, I don't know if bucket is relevant. It is simply a mechanism to host the sam build artifact for actual deployment.

File: src/lambda/commands/deploySamApplication.ts [hunk 4]
```diff
@@ -26,19 +33,34 @@ export async function deploySamApplication(
...
+        restParams.outputChannel.appendLine(msg)
         // TODO: Is this the right way to provide this feedback?
-        vscode.window.showInformationMessage(`Successfully deployed '${stackName}' to ${s3Bucket}`)
+        vscode.window.showInformationMessage(msg)
     } catch (err) {
+        // TODO: Add nls support
+        const msg = `Failed ${stage} "${stackName}" using S3 bucket "${s3Bucket}": ${String(err)}`
```
Comment (awschristou) [inline src/lambda/commands/deploySamApplication.ts]: Minor: Consider: 
`Failed to deploy SAM Application. An error occurred during stage ${stage}: ${String(err)}`

The lines above can provide the stack/bucket info

File: src/lambda/commands/deploySamApplication.ts (hunk 1 above)
Comment (micholas) [reply src/lambda/commands/deploySamApplication.ts]: ✔

File: src/lambda/commands/deploySamApplication.ts (hunk 2 above)
Comment (micholas) [reply src/lambda/commands/deploySamApplication.ts]: ✔

File: src/lambda/commands/deploySamApplication.ts (hunk 4 above)
Comment (micholas) [reply src/lambda/commands/deploySamApplication.ts]: ✔

File: src/lambda/commands/deploySamApplication.ts (hunk 3 above)
Comment (micholas) [reply src/lambda/commands/deploySamApplication.ts]: ✔

**Pull request merged**